### PR TITLE
Integrate with kotlin gradle plugin

### PR DESCRIPTION
**Pull request by zpencer**
This tries to be generic wrt language build tasks, and specifically tests against kotlin. If users want to enable additional unsupported languages, they can set the project property `protobuf_gradle_plugin_additional_languages` as a `List<String>` of more languages.

Test projects are now created programmatically by repeatedly copying overlay directories into a temp directory.
Updated tests and examples:
 - testProjectKotlin is a kotlin only project
 - testProjectJavaAndKotlin is a java+kotlin project

This only tests a plain kotlin project. A follow up PR will address Android.

### CODE REVIEW

File: src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy [hunk 1]
```diff
@@ -409,12 +427,22 @@ class ProtobufPlugin implements Plugin<Project> {
           }
         }
       } else {
-        project.sourceSets.each { sourceSet ->
-          Task javaCompileTask = project.tasks.getByName(sourceSet.getCompileTaskName("java"))
-          project.protobuf.generateProtoTasks.ofSourceSet(sourceSet.name).each { generateProtoTask ->
-            javaCompileTask.dependsOn(generateProtoTask)
-            generateProtoTask.getAllOutputDirs().each { dir ->
-              javaCompileTask.source project.fileTree([dir:dir])
+        project.sourceSets.each { SourceSet sourceSet ->
```
Comment (zhangkun83) [inline src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy]: Do you have plan to add support for Android projects?

File: src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy [hunk 2]
```diff
@@ -59,6 +60,23 @@ class ProtobufPlugin implements Plugin<Project> {
             'android-library',
     ]
 
+    private static final String USER_LANG_PROP = 'protobuf_gradle_plugin_additional_languages'
```
[the remark this answers is not in the document]
Comment (zpencer) [reply src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy]: Fixed

File: src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy (hunk 1 above)
Comment (zpencer) [reply src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy]: Yes, I have a PR that's ready to be posted after this one is merged.

**Comment by zpencer** [comment]
Since https://github.com/google/protobuf-gradle-plugin/pull/208 is now resolved, I think we can release what we have on master branch @zhangkun83

**Pull request merged**